runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -191,21 +197,56 @@ private Ellipse2D drawEllipse(Graphics2D graphics, int x, int y)
 
 	private void drawSkillImage(Graphics2D graphics, XpGlobe xpGlobe, int x, int y)
 	{
-		BufferedImage skillImage = iconManager.getSkillImage(xpGlobe.getSkill());
-
-		if (skillImage == null)
+		BufferedImage toDraw;
+		if (xpGlobe.getSkillIcon() != null && xpGlobe.getGlobeSize() == config.xpOrbSize())
 		{
-			return;
+			toDraw = xpGlobe.getSkillIcon();
+		}
+		else
+		{
+			toDraw = generateScaledIcon(xpGlobe.getSkill());
+			if (toDraw == null)
+			{
+				return;
+			}
+			xpGlobe.setSkillIcon(toDraw);
+			xpGlobe.setGlobeSize(config.xpOrbSize());
 		}
 
 		graphics.drawImage(
-			skillImage,
-			x + (config.xpOrbSize() / 2) - (skillImage.getWidth() / 2),
-			y + (config.xpOrbSize() / 2) - (skillImage.getHeight() / 2),
+			toDraw,
+			x + (config.xpOrbSize() / 2) - (toDraw.getWidth() / 2),
+			y + (config.xpOrbSize() / 2) - (toDraw.getHeight() / 2),
 			null
 		);
 	}
 
+	private BufferedImage generateScaledIcon(Skill skill)
```

Comment:
Seems like this can just be an Image class removing the need to create a BufferedImage and drawing to it.

Reply:
It is preferred to draw to a BufferedImage because it only has to scale once. Simply returning the result of `getScaledInstance` makes it rescale everytime the image is drawn.

Reply:
Didn't realize that was the behavior, that makes sense then.